projects/packages/videopress/src/client/state/reducers.js

```diff
@@ -124,6 +126,72 @@ const videos = ( state, action ) => {
 			};
 		}
 
+		case SET_VIDEO_PRIVACY: {
+			const { id, privacySetting } = action;
+			const items = [ ...( state.items ?? [] ) ];
+			const videoIndex = items.findIndex( item => item.id === id );
+
+			if ( videoIndex < 0 ) {
+				return state;
+			}
+
+			// current -> previous value of privacy
+			const current = items[ videoIndex ].privacySetting;
+
+			// Set privacy setting straigh in the state. Let's be optimistic.
+			items[ videoIndex ] = {
+				...items[ videoIndex ],
+				privacySetting,
+			};
+
+			// Set metadata about the privacy change.
+			const _metaItems = { ...( state._meta?.items ?? [] ) };
+			const _metaVideo = _metaItems[ id ] ?? {};
+
+			return {
+				...state,
+				items,
+				_meta: {
+					...state._meta,
+					items: {
+						..._metaItems,
+						[ id ]: {
+							..._metaVideo,
+							isUpdatingPrivacy: true,
+							hasBeenUpdatedPrivacy: false,
+							prevPrivacySetting: current,
+						},
+					},
+				},
+			};
+		}
+
+		case UPDATE_VIDEO_PRIVACY: {
+			const { id } = action;
+
+			const _metaItems = { ...( state._meta?.items ?? [] ) };
+			if ( ! _metaItems?.[ id ] ) {
+				return state;
+			}
+
+			const _metaVideo = _metaItems[ id ] ?? {};
+
+			return {
+				...state,
+				_meta: {
+					...state._meta,
+					items: {
+						..._metaItems,
+						[ id ]: {
+							..._metaVideo,
+							isUpdatingPrivacy: false,
+							hasBeenUpdatedPrivacy: true,
```

Comment:
Shouldn't `prevPrivacySetting` be set to `null`?

Reply:
yes, makes sense. Updating...